streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1291,29 +1293,77 @@ private void resetOffsets(final Set<TopicPartition> partitions, final Exception
         final Set<String> loggedTopics = new HashSet<>();
         final Set<TopicPartition> seekToBeginning = new HashSet<>();
         final Set<TopicPartition> seekToEnd = new HashSet<>();
+        final Map<TopicPartition, Duration> seekByDuration = new HashMap<>();
         final Set<TopicPartition> notReset = new HashSet<>();
 
         for (final TopicPartition partition : partitions) {
-            final AutoOffsetResetStrategy offsetResetStrategy = topologyMetadata.offsetResetStrategy(partition.topic());
+            final Optional<AutoOffsetResetStrategy> offsetResetStrategy = topologyMetadata.offsetResetStrategy(partition.topic());
 
             // This may be null if the task we are currently processing was apart of a named topology that was just removed.
             // TODO KAFKA-13713: keep the StreamThreads and TopologyMetadata view of named topologies in sync until final thread has acked
```

Comment:
can we update this TODO to make it about removing the null check once we remove named topologies

Reply:
Sure, I can update it to be more explicit -- I do read it as "remove null check after NT are removed" already though

Reply:
Could do this in a follow up PR, too, to avoid waiting for CI again...

Reply:
was less concerned about us misinterpreting the TODO, more that it just seems weird to have a TODO in the code for a ticket we never plan to do (though that reminds me, I should go close that ticket)